src/format.rs

```diff
@@ -224,9 +230,12 @@ async fn run(
         let start_row = &buf[(ctx.codeblock_start as i32 + offset) as usize];
         let whitespace = utils::get_start_whitespace(start_row);
 
+        let indent = utils::get_start_whitespace(&buf[(ctx.start as i32) as usize]);
```

Comment:
You need to keep the offset in mind.

```suggestion
        let indent = utils::get_start_whitespace(&buf[(ctx.start as i32 + offset) as usize]);
```